agent/utils/reference/reference.go

```diff
@@ -78,3 +78,34 @@ func GetDigestFromRepoDigests(repoDigests []string, imageRef string) (digest.Dig
 
 	return "", fmt.Errorf("found no repo digest matching '%s'", imageRef)
 }
+
+// Given an image reference and a manifest digest string, returns a canonical reference
+// for the image.
+// If the image reference has a digest then the canonical reference will still use the provided
+// manifest digest overwriting the exiting digest in the image reference.
```

Comment:
nit (non-blocking): typo in `exiting`?

Reply:
Will fix this in a follow-up PR right after.